File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    static bool checker(vector<int>a,vector<int>b)
    {
        return a[0] < b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>>mergedIntervals;
        sort(intervals.begin(),intervals.end(),checker);
        int i = 0;
        int n = intervals.size();
        while(i < n)
        {
            int start = intervals[i][0];
            int end = intervals[i][1];
            int j = i;
            while(j < intervals.size() && intervals[j][0] <= end)
            {
                end = max(end,intervals[j][1]);
                j++;
            }
            i = j;
            mergedIntervals.push_back({start,end});
        }
        return mergedIntervals;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        initial_start = -1
        initial_end = -1
        res = []
        print(intervals)
        for i in range(len(intervals)):
            if initial_start == -1 and initial_end == -1:
                initial_start = intervals[i][0]
                initial_end = intervals[i][1]
            else:
                if intervals[i][0] <= initial_end:
                
                    initial_end = max(initial_end,intervals[i][1])
                else:
                    res.append([initial_start,initial_end])
                    initial_start = intervals[i][0]
                    initial_end = intervals[i][1]
        res.append([initial_start,initial_end])
        return res